src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -222,7 +222,6 @@ private void setPageRangeAttribute(int from, int to, boolean isRangeSet) {
             // 1 less than pageRanges attribute
             if (isRangeSet) {
                 attributes.add(new PageRanges(from+1, to+1));
-                attributes.add(SunPageSelection.RANGE);
```

Comment:
why was this attribute added, and why is it being removed now? is the bug in SunPageSelection?

Reply:
> why was this attribute added, and why is it being removed now? is the bug in SunPageSelection?

I am not sure why this attribute added, but noticed for other OS's this workflow will be taken care by _RasterPrinterJob_ . With this attribute application pass through range and ended up in invalid page printing range.